cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -402,4 +404,19 @@ private InputFormatProvider createInputFormatProvider(String datasetName,
                                                         @Nullable List<Split> splits) {
     return createInput((Input.DatasetInput) Input.ofDataset(datasetName, datasetArgs, splits)).getInputFormatProvider();
   }
+
+  @Nullable
+  private MetricsContext getWorkflowMetricsContext(@Nullable WorkflowProgramInfo workflowProgramInfo, Program program,
```

Comment:
the call statement gets verbose, can i keep it this way ? as it's easier to read this way.